core/client/verify.go

```diff
@@ -110,53 +111,52 @@ func (v *Verifier) VerifyGetEntryResponse(ctx context.Context, domainID, appID,
 		nonce := in.GetCommitted().GetKey()
 		if err := commitments.Verify(userID, appID, commitment, data, nonce); err != nil {
 			Vlog.Printf("✗ Commitment verification failed.")
-			return fmt.Errorf("commitments.Verify(%v, %v, %v, %v, %v): %v", userID, appID, commitment, data, nonce, err)
+			return nil, nil, fmt.Errorf("commitments.Verify(%v, %v, %v, %v, %v): %v", userID, appID, commitment, data, nonce, err)
 		}
 	}
 	Vlog.Printf("✓ Commitment verified.")
 
 	index, err := v.Index(in.GetVrfProof(), domainID, appID, userID)
 	if err != nil {
 		Vlog.Printf("✗ VRF verification failed.")
-		return err
+		return nil, nil, err
 	}
 	Vlog.Printf("✓ VRF verified.")
 
 	leafProof := in.GetLeafProof()
 	if leafProof.GetLeaf() == nil {
-		return ErrNilProof
+		return nil, nil, ErrNilProof
 	}
 	leafProof.Leaf.Index = index[:]
 
-	if err := v.mapVerifier.VerifyMapLeafInclusion(in.GetSmr(), leafProof); err != nil {
-		Vlog.Printf("✗ Sparse tree proof verification failed.")
-		return err
-	}
-	Vlog.Printf("✓ Sparse tree proof verified.")
-
-	if err := v.mapVerifier.VerifySignedMapRoot(in.GetSmr()); err != nil {
+	mapRoot, err := v.mapVerifier.VerifySignedMapRoot(in.GetSmr())
+	if err != nil {
 		Vlog.Printf("✗ Signed Map Head signature verification failed.")
-		return fmt.Errorf("VerifySignedMapRoot(): %v", err)
+		return nil, nil, fmt.Errorf("VerifySignedMapRoot(): %v", err)
 	}
+
 	Vlog.Printf("✓ Signed Map Head signature verified.")
+	if err := v.mapVerifier.VerifyMapLeafInclusion(mapRoot, leafProof); err != nil {
+		Vlog.Printf("✗ Sparse tree proof verification failed.")
+		return nil, nil, fmt.Errorf("VerifyMapLeafInclusion(): %v", err)
+	}
+	Vlog.Printf("✓ Sparse tree proof verified.")
```

Comment:
Nice idea with the checkmark/cross!

Reply:
UTF-8 source code!

Reply:
Yeah, I meant the cross above in line 140 :)